src/frontend/lib/mapeo-schema.d.ts

```diff
@@ -0,0 +1,189 @@
+// Refer to https://github.com/digidem/mapeo-schema/blob/master/docs/README.md
+declare module "mapeo-schema" {
+  // TODO: do we need this?
+  interface Common {
+    created_at: string;
+    deviceId?: string;
+    id: string;
+    links?: string[];
+    schemaVersion: number;
+    timestamp?: string;
+    userId?: string;
+    type: string;
+  }
+
+  export interface Position {
+    coords?: {
+      accuracy?: number;
+      altitude?: number;
+      heading?: number;
+      latitude?: number;
+      longitude?: number;
+      speed?: number;
+    };
+    mocked?: boolean;
+  }
+
+  export interface Observation {
+    attachments?: {
+      id: string;
+      type?: string;
+    }[];
+    created_at: string;
+    deviceId?: string;
+    id: string;
+    lat?: number | null;
+    links?: string[];
+    lon?: number | null;
+    metadata?: {
+      lastSavedPosition?: Position;
+      manualLocation?: boolean;
+      position?: Position;
+      positionProvider?: {
+        gpsAvailable?: boolean;
+        locationServicesEnabled?: boolean;
+        networkAvailable?: boolean;
+        passiveAvailable?: boolean;
+      };
+    };
+    refs?: {
+      id: string;
+    }[];
+    schemaVersion: number;
```

Comment:
@gmaclennan based on https://github.com/digidem/mapeo-mobile/pull/723/commits/3f2c612b2696b953e1913cdabc9e3131d361d8da#r697847676, should this be `4` then? Seems like it should based on https://github.com/digidem/mapeo-schema/blob/master/docs/observation.md#schemaversion

Reply:
Yes. The `mapeo-schema` repo exports Flow types (which are generated from the JSON Schema). The goal is to have a single source of truth for types/schema. The `mapeo-schema` repo should probably be updated to export TypeScript rather than Flow. One option that works well is https://github.com/sinclairzx81/typebox. Another option is to use https://jsontypedef.com instead of Json Schema, which maps more easily to Typescript, but does not support everything that Json Schema does.